vllm/attention/layer.py

```diff
@@ -49,13 +49,36 @@ def __init__(
         use_mla: bool = False,
         prefix: str = "",
         attn_type: str = AttentionType.DECODER,
+        kv_sharing_target_layer_name: Optional[str] = None,
         **extra_impl_args,
     ) -> None:
         """
         The KV cache is stored inside this class and is accessed via
         `self.kv_cache`.
         """
         super().__init__()
+
+        if kv_sharing_target_layer_name is not None:
+            if not envs.VLLM_USE_V1:
+                raise NotImplementedError(
+                    "Cross-layer KV sharing is not supported in V0.")
+
+            # Verify target layer is valid for cross-layer KV sharing
+            from vllm.model_executor.models.utils import extract_layer_index
+            current_layer_idx = extract_layer_index(prefix)
+            target_layer_idx = extract_layer_index(
+                kv_sharing_target_layer_name)
+            if current_layer_idx <= target_layer_idx:
+                comp_str = ("is equal to" if current_layer_idx
+                            == target_layer_idx else "comes after")
```

Comment:
Can you simplify the logic here? I think we don't need to distinguish between "is equal to" and "comes after".